ibis/mapd/client.py

```diff
@@ -152,6 +170,32 @@ def to_df(self):
         if isinstance(self.cursor, Cursor):
             col_names = [c.name for c in self.cursor.description]
             result = pd.DataFrame(self.cursor.fetchall(), columns=col_names)
+
+            if not GEODATAFRAME_SUPPORTED:
```

Comment:
Instead of mixing both kinds of DataFrame return values in a single method, how about making two different kinds of cursor classes that both implement a common method (say `to_df`), and wherever the cursor is instantiated branch on `GEODATAFRAME_SUPPORTED` and instantiate the appropriate cursor. Better yet, make a `dict`:

```python
cursors = {
    True: GeoOmniSciCursor,
    False: DefaultOmniSciCursor,
}
```

Reply:
sounds good. I will change that. thanks!